telemetry/prometheus/metrics.go

```diff
@@ -279,3 +270,58 @@ func setDefaultCfg(cfg *Config) {
 		cfg.HistogramBucketCount = DefaultBucketCount
 	}
 }
+
+// Helper method to get or create a GaugeVec.
+func (p *metrics) mustGetOrCreateGaugeVec(name string, labels []string) *prometheus.GaugeVec {
```

Comment:
IMO this is the best refactor we can do to avoid duplicate code.

While there are still duplicate code in this and the following methods, but they are of different types, and Prometheus itself is lack of a generic type - so further refactoring requires extra enum/params/swtich statement - which is not very beneficial.